src/clojure/nrepl/misc.clj

```diff
@@ -6,13 +6,24 @@
   (:require [clojure.java.io :as io]))
 
 (defn log
-  [ex & msgs]
-  (let [ex (when (instance? Throwable ex) ex)
-        msgs (if ex msgs (cons ex msgs))]
+  [ex-or-msg & msgs]
+  (let [ex (when (instance? Throwable ex-or-msg) ex-or-msg)
+        msgs (if ex msgs (filter identity (cons ex-or-msg msgs)))]
     (binding [*out* *err*]
       (apply println "ERROR:" msgs)
       (when ex (.printStackTrace ^Throwable ex)))))
 
+(defmacro noisy-future
```

Comment:
Not sure that's the best name for this function. :D

Reply:
Heh, happy to change it if you like - any preferred name.  And not positive you'd want to use it as pervasively as the commit does.  I just added it after being thwarted by some hidden exceptions when I was working on the sockets.